proposals/NNNN-improve-starting-streaming-services.md

```diff
@@ -0,0 +1,54 @@
+
+If the app enters a streamable state (reentered `HMI_FULL` and/or `OnHMIStatus.videoStreamingState` is `STREAMABLE`) the libraries start the video manager first followed by the audio manager. If the video manager received NAK for the video service, the library should not start the audio manager's audio service. Instead the library should retry the video service after two seconds of waiting and hold back the audio manager from starting. No other `StarService` request should be send during this time. the infinite retry method is justified due to the app being in a streamable state (HMI_FULL etc.). No other application would be interfering during this time unless the app leaves the streamable state. 
+
+The library should retry until the app leaves the streamable state. If the app left the streamable state during the two seconds, the library must prevent sending the `StartService` request and stop the retry process.
+
+## Potential downsides
+
+In certain situations where the retry mechanism occurs, the proposed solution can increase the time until the app has started the video and audio stream for up to two seconds. However, the libraries gain a lot of stability for error cases which is much more valuable.
+
+It can be possible that this retry mechanism puts some stress to existing infotainment systems as they were never tested against this method. However, the current behavior of the libraries to stop after one failure would leave the user in an unacceptable state with an non-functional app. Therefore a retry mechanism can only improve the user experience.
+
+## Impact on existing code
+
+The proposal doesn't change when the video or audio manager becomes streamable. The library should still listen for SDL Core information when the app becomes streamable. This proposal only synchronizes `StartService` requests and adds a retry mechanism to overcome failed attempts.
+
+All libraries require changes on the video and audio managers. Instead of actively sending `StartService` requests both managers have to wait for another source to trigger the starting routine. The implementation details are up to the project maintainers and the implementors. The iOS library already contains a general streaming manager called `SDLStreamingMediaManager` which could take the role of starting the video and audio manager. A similar manager could be added to the Java Suite taking over the responsibilities from the SDL manager.
```

Comment:
```suggestion
All app libraries require changes on the video and audio managers. Instead of actively sending `StartService` requests, both managers have to wait for another source to trigger the starting routine. The implementation details are up to the SDLC Project Maintainer and the implementers. The iOS library already contains a general streaming manager called `SDLStreamingMediaManager` which could take the role of starting the video and audio manager. A similar manager could be added to the Java Suite and JavaScript Suite libraries, taking over the responsibilities from the SDL manager.
```